### PR TITLE
feat(executions): Adding new ADMIN endpoint to import execution

### PR DESCRIPTION
**Pull request by srekapalli**
Adding new Admin endpoint to import execution from another spinnaker instance.

Useful to get a execution into other envs for troubleshooting.
Only imports executions with TERMINAL, SUCCEEDED or CANCELLED states.

### CODE REVIEW

File: orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/AdminController.groovy [hunk 1]
```diff
@@ -74,6 +79,23 @@ class AdminController {
     forceExecutionCancellationCommand.forceCancel(executionType, executionId, canceledBy)
   }
 
+
+  @RequestMapping(value = "/executions/", method = RequestMethod.POST)
+  @ResponseStatus(HttpStatus.OK)
+  void createExecution(@RequestBody Execution execution) {
+
+    if (execution.status in [ExecutionStatus.CANCELED, ExecutionStatus.SUCCEEDED, ExecutionStatus.TERMINAL]) {
```
Comment (marchello2000) [inline orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/AdminController.groovy]: do you think it's worth validating that the app & pipeline exist. Maybe at least spitting our an http response with some text:
```
Imported execution with ID: {} for app: {}, pipeline: {} with {n} stages
Note: the app doesn't exist in this environment
```

I think it would be valuable otherwise folks might import it and not see it and be confused?
Also, would be useful to know the ID of newly imported execution (though it's the same as the payload)

**Comment by marchello2000** [comment]
Thanks for doing this, I think it would be useful and LGTM, but would like another opinion as to any side effects this could cause, @cfieber / @robzienert ?

File: orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/AdminController.groovy [hunk 2]
```diff
@@ -74,6 +87,35 @@ class AdminController {
     forceExecutionCancellationCommand.forceCancel(executionType, executionId, canceledBy)
   }
 
+
+  @PostMapping(value = "/executions/")
+  @ResponseStatus(HttpStatus.CREATED)
+  Map<String, String> createExecution(@RequestBody Execution execution) {
+
+    if (front50Service && !front50Service.get(execution.application)) {
```
Comment (ajordens) [inline orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/AdminController.groovy]: I believe `front50Service.get()` will raise an exception if the application doesn't exist (vs. returning null).

I'm not sure we want to import executions for applications that don't exist?

File: orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/AdminController.groovy [hunk 3]
```diff
@@ -74,6 +87,35 @@ class AdminController {
...
+      log.warn('No application exists with name: ' + execution.application)
+    }
+
+    try {
+      executionRepository.retrieve(execution.type, execution.id)
+      log.warn('Execution found with id: []', execution.id)
```
Comment (ajordens) [inline orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/AdminController.groovy]: `warn` seems like the wrong logging level? `debug` maybe? (or perhaps it's not even relevant given it's going to raise an exception immediately)

File: orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/AdminController.groovy [hunk 4]
```diff
@@ -74,6 +87,35 @@ class AdminController {
...
+      throw new InvalidRequestException('Execution already exists with id: ' + execution.id)
+    } catch(ExecutionNotFoundException e) {
+      log.info('Execution not found .. can import it..')
```
Comment (ajordens) [inline orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/AdminController.groovy]: This log message seems irrelevant and would get lost in the noise? There's no identifier included in the output.
